ws/src/session.rs

```diff
@@ -97,15 +97,15 @@ impl LivenessPoll {
 
 		let (index, rx) = {
 			let mut task_slab = task_slab.lock();
-			if !task_slab.has_available() {
+			if task_slab.len() == task_slab.capacity() {
```

Comment:
Seems that the slab is now being resized everytime you `insert` and there is no available space left. We could get rid of this and just rely on slab re-allocating internally.

Reply:
That's true, but slab's internal mechanisms seems to be growing it linearly, while we're using exponential size growth here — might be better, since slab reallocations are ineffective (and there's even a reallocation warning in [the doc](https://docs.rs/slab/0.4.0/slab/index.html#capacity-and-reallocation).)

I'm all in favor of more simple solution — but I wasn't able to tell if our expectations would or would not be violated in this case.

Reply:
Fair points. Looks good then.